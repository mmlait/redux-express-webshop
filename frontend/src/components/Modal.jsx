import React from 'react';
import styled from 'styled-components';
import colors from '../colors';

const ModalWrapper = styled.div`
  background-color: ${colors.modalBgOverlay};
  position: fixed;
  z-index: 1;
  padding-top: 100px;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
`

const ModalContent = styled.div`
  display: flex;
  flex-direction: column;
  margin: auto;
  padding: 10px;
  width: 90%;
  @media (min-width: 350px) {
    width: 350px;
  }
  @media (min-width: 900px) {
    width: 450px;
  }
`

const Modal = ({content}) => {
  return (
    <ModalWrapper>
      <ModalContent>
        {content}
      </ModalContent>
    </ModalWrapper>
  );
};

export default Modal;