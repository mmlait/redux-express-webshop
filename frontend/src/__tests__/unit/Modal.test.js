import React from 'react';
import { create } from 'react-test-renderer'
import { mount } from 'enzyme';
import Modal from '../../components/Modal';

let props = {
  content: 'Test content'
}

describe('Modal component', () => {
  it('should render', () => {
    let tree = create(<Modal />)
    expect(tree.toJSON()).toMatchSnapshot();
  })
  it ('should render with props', () => {
    const wrapper = mount(
      <Modal content={props.content} />
    );
    expect(wrapper.props().content).toEqual(props.content);
  })
})