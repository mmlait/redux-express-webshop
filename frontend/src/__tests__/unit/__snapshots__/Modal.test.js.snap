// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should render 1`] = `
<div
  className="sc-bdVaJa iYMbCT"
>
  <div
    className="sc-bwzfXH geHczq"
  />
</div>
`;
